blame.c

```diff
@@ -1423,10 +1423,17 @@ static struct blame_origin *find_rename(struct repository *r,
 	struct blame_origin *porigin = NULL;
 	struct diff_options diff_opts;
 	int i;
+	extern int rename_detection_mode;
```

Comment:
No, this won't work. You declared it as `static` in `builtin/blame.c`, which makes it file-local, and you cannot ever see it from this here file.

Instead, you _have_ to introduce a new attribute in `blame.h`, probably in `struct blame_scoreboard` next to the `xdl_opts` (which are also diff-related).